saq/queue/postgres.py

```diff
@@ -0,0 +1,334 @@
+"""
+Postgres Queue
+"""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import time
+import typing as t
+
+from psycopg.types.json import Jsonb
+from psycopg_pool import AsyncNullConnectionPool, AsyncConnectionPool
+
+from saq.job import (
+    Job,
+    Status,
+)
+from saq.queue.base import Queue, logger
+from saq.utils import now
+
+if t.TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from saq.types import (
+        CountKind,
+        ListenCallback,
+        DumpType,
+        LoadType,
+        QueueInfo,
+        QueueStats,
+    )
+
+JOBS_TABLE = "saq_jobs"
+
+CREATE_JOBS_TABLE = """
```

Comment:
1. Some applications might not have permission to create tables, so this should probably be made optional and executable by a standalone process
2. Should we be thinking ahead about migrations? What if we want to add a column, etc. Perhaps we should be syncing a sequence of changes.

Reply:
Backwards compatible migrations, that don’t require users to go in and alter their DB, that is

Reply:
i think it's ok that we always create the table, but i'd make it a tad more configurable like what the table name is?

Reply:
So if we ever need to change we expect users to update the table name?